### DISCUSSION TITLE
How can I add Anotations to the PDF?

### DISCUSSION THREAD

**Discussion by cris9600** [wojtekmaj/react-pdf#1718] [Q&A]
Hello, i'm trying to make a pdf viewer with anotations, like  drawing,  marking text, adding text, etc.

I see that when the components render the PDF  using "<Document><Page></Page></Document>" it has 3 layers:
1. The "image layer " of the pdf page.
2. The "text layer".
3. The "anotation layer".

**I would like to know how can I achieve this, adding anotations to the pdf, where should i add them, and  if posible to download the new pdf with anotations.**

Here I render the pdf: 
```
  <Document
    file={{ url: file }}
    onLoadSuccess={onDocumentLoadSuccess}
    inputRef={documentRef} 
  >
    {
    Array.from(new Array(numPages), (el, index) => (
        <Page
          key={`page_${index + 1}`}
          scale={1.5}
          pageNumber={index + 1}
          renderTextLayer={true}
        >
        </Page>
    ))
  }
   </Document>
```

**Comment by wojtekmaj** [comment]
react-pdf does not allow this much interactivity unfortunately. It's mostly for viewing PDFs. Sorry!

**Reply by cris9600** [reply]
Thanks for your answer. Even though the annotations cannot be done directly, this component helps me a lot.
